bundles/org.openhab.binding.revogi/src/main/java/org/openhab/binding/revogi/internal/udp/DatagramSocketWrapper.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.revogi.internal.udp;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.SocketException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * The {@link DatagramSocketWrapper} wraps Java's DatagramSocket for better testing
+ * UdpSenderService
+ *
+ * @author Andi Bräu - Initial contribution
+ */
+@NonNullByDefault
+public class DatagramSocketWrapper {
+
+    @Nullable
+    private DatagramSocket datagramSocket;
+
+    public void initSocket() throws SocketException {
+        closeSocket();
+        datagramSocket = new DatagramSocket();
+        datagramSocket.setBroadcast(true);
+        datagramSocket.setSoTimeout(3);
+    }
+
+    public void closeSocket() {
+        if (datagramSocket != null && !datagramSocket.isClosed()) {
+            datagramSocket.close();
```

Comment:
`datagramSocket` needs to be stored to a local variable to fix the compiler warning. Same for below.

Reply:
done, even if I don't understand why that's a problem for the compiler. I still need to male a null check for the local variable. So in my opinion it's just more boilerplate code to write.

Reply:
The compiler complains because `datagramSocket` is a field and could theoretically be modified by another thread exactly between the null check and the access in the next line.